packages/core/saved-objects/core-saved-objects-api-server/src/saved_objects_client.ts

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObject } from '@kbn/core-saved-objects-common';
+import type {
+  SavedObjectsBaseOptions,
+  SavedObjectsFindOptions,
+  SavedObjectsClosePointInTimeOptions,
+  SavedObjectsOpenPointInTimeOptions,
+  SavedObjectsCreatePointInTimeFinderOptions,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+  SavedObjectsUpdateObjectsSpacesObject,
+  SavedObjectsUpdateObjectsSpacesOptions,
+  SavedObjectsCollectMultiNamespaceReferencesObject,
+  SavedObjectsUpdateObjectsSpacesResponse,
+  SavedObjectsResolveResponse,
+  ISavedObjectsPointInTimeFinder,
+  SavedObjectsRemoveReferencesToOptions,
+  SavedObjectsClosePointInTimeResponse,
+  SavedObjectsDeleteOptions,
+  SavedObjectsCreatePointInTimeFinderDependencies,
+  SavedObjectsCheckConflictsResponse,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsCollectMultiNamespaceReferencesResponse,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsFindResponse,
+  SavedObjectsCreateOptions,
+  SavedObjectsCheckConflictsObject,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkResolveResponse,
+  SavedObjectsBulkResolveObject,
+  SavedObjectsBulkGetObject,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsOpenPointInTimeResponse,
+  SavedObjectsRemoveReferencesToResponse,
+  SavedObjectsCollectMultiNamespaceReferencesOptions,
+  SavedObjectsBulkResponse,
+} from './apis';
+
+/**
+ * Saved Objects is Kibana's data persisentence mechanism allowing plugins to
+ * use Elasticsearch for storing plugin state.
+ *
+ * ## SavedObjectsClient errors
```

Comment:
I'm on the fence about keeping the docs on SO errors here. On the one hand, any SO client method could return an error but on the other, perhaps it should live with the Response types. 
Then again, duplicating the docs for every API would be overkill, so let's keep it here for now.

Reply:
Our public SO tsdoc could totally benefit from some re-thinking for sure. I didn't want to open this question during this PR though 😅